regcore/db/django_models.py

```diff
@@ -1,45 +1,98 @@
 """Each of the data structures relevant to the API (regulations, notices,
 etc.), implemented using Django models"""
+import collections
+
 from django.core.exceptions import ObjectDoesNotExist
 
 from regcore.db import interface
 from regcore.models import Diff, Layer, Notice, Regulation
 
 
+def treeify(node, tree_id, pos=1, level=0):
+    """Set tree properties in memory.
+    """
+    node['tree_id'] = tree_id
+    node['level'] = level
+    node['left'] = pos
+    for child in node.get('children', []):
+        pos = treeify(child, tree_id, pos=pos + 1, level=level + 1)
+    pos = pos + 1
+    node['right'] = pos
+    return pos
+
+
+def get_adjacency_map(regs):
+    """Build mapping from node IDs to child records
+    :param regs: List of `Regulation` records
+    """
+    ret = collections.defaultdict(list)
+    for reg in regs:
+        if reg.parent_id is not None:
+            ret[reg.parent_id].append(reg)
+    return ret
+
+
+def get_id(reg, version):
+    return '{}-{}'.format(version, '-'.join(reg['label']))
```

Comment:
I'd recommend we use something other than a dash to delimit the version and label. It's very unlikely we'd see a collision with the dashes within the label, but it couldn't hurt to use a (say) colon here.

Very, very minor, but this (and `get_adjacency_map`) use "get" when they probably mean "build" or "construct"